### PR TITLE
Refactor api keys and naming

### PR DESCRIPTION
**Pull request by neubig**
# Description

This rather large commit has two main goals:
1. Remove all "api_key" variables and manage API keys solely through environment variables.
2. Remove many mentions to OpenAI where they are not necessary.

# References

- Fixes https://github.com/neulab/prompt2model/issues/330
- Fixes https://github.com/neulab/prompt2model/issues/332
- Also relevant to https://github.com/neulab/prompt2model/issues/310

# Blocked by

- NA

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ pip install prompt2model
 To use OpenAI, please follow these
 
 - Sign up on the OpenAI website and obtain an
```
Comment (zhaochenyang20) [inline README.md:37]: I guess it's not "OpenAI website"? 🤔

File: prompt2model/dataset_generator/__init__.py [hunk 2]
```diff
@@ -1,10 +1,10 @@
 """Import DatasetGenerator classes."""
 from prompt2model.dataset_generator.base import DatasetGenerator, DatasetSplit
 from prompt2model.dataset_generator.mock import MockDatasetGenerator
-from prompt2model.dataset_generator.openai_gpt import OpenAIDatasetGenerator
+from prompt2model.dataset_generator.prompt_based import PromptBasedDatasetGenerator
```
Comment (zhaochenyang20) [inline prompt2model/dataset_generator/__init__.py:4]: APIBasedDatasetGenerator?

File: prompt2model/dataset_generator/prompt_based.py [hunk 3]
```diff
@@ -17,14 +16,14 @@
 from tqdm import tqdm
 
 from prompt2model.dataset_generator.base import DatasetGenerator, DatasetSplit
-from prompt2model.dataset_generator.openai_gpt_template import construct_meta_prompt
+from prompt2model.dataset_generator.prompt_based_template import construct_meta_prompt
```
Comment (zhaochenyang20) [inline prompt2model/dataset_generator/prompt_based.py:19]: prompt_template

File: prompt2model/dataset_generator/prompt_based.py [hunk 4]
```diff
@@ -81,29 +77,22 @@ def __init__(
             cache_root: The root directory for caching generated examples.
 
         Raises:
-            ValueError: If an API key is not provided and set as an environment
-            variable, or if the 'max_api_calls' value is not greater than 0.
+            ValueError: If the 'max_api_calls' value is not greater than 0.
 
         Note:
-            For the OpenAI GPT-3.5 API, Temperature ranges from 0 to 2. Higher
+            Temperature ranges from 0 to 2. Higher
```
Comment (zhaochenyang20) [inline prompt2model/dataset_generator/prompt_based.py:83]: I just want to ensure that every model that `lightllm` supports can change the temperature.

File: prompt2model/dataset_generator/prompt_based.py [hunk 5]
```diff
@@ -442,7 +402,7 @@ def create_all_examples_dataset_and_generated_dataset(
     def compute_batch_size(
         self, expected_num_examples: int, generated_dataset: Dataset
     ) -> int:
-        """Computes the batch size for OpenAI API calls in a batch.
+        """Computes the batch size for API calls in a batch.
```
Comment (zhaochenyang20) [inline prompt2model/dataset_generator/prompt_based.py:405]: Also, do these models support changing the `n` parameters?

File: prompt2model/dataset_generator/readme.md [hunk 6]
```diff
@@ -5,26 +5,21 @@
...
-```python
-api_key = "<your-api-key>"
-dataset_generator = OpenAIDatasetGenerator(api_key)
-```
-
-Or, set as an environment variable:
+Set an API key as an environment variable:
 
 ```bash
 export OPENAI_API_KEY="<your-api-key>"
```
Comment (zhaochenyang20) [inline prompt2model/dataset_generator/readme.md:25]: This should not be `OPENAI_API_KEY`.

File: prompt2model/utils/api_tools.py [hunk 7]
```diff
@@ -15,7 +15,7 @@
 from litellm import acompletion, completion
 from tqdm.asyncio import tqdm_asyncio
 
-OPENAI_ERRORS = (
+API_ERRORS = (
```
Comment (zhaochenyang20) [inline prompt2model/utils/api_tools.py:18]: I am pretty confused. If I use API to call Claude and this API call fails, will it return me an error with `openai.error`? Will it be `claude.error`?

**Review by zhaochenyang20 — CHANGES_REQUESTED** [review]
Final quesiton. I don't find where the API key is set to use `ligthtLLM`.

File: README.md (hunk 1 above)
Comment (neubig) [reply README.md:37]: Clarified.

File: prompt2model/dataset_generator/__init__.py (hunk 2 above)
Comment (neubig) [reply prompt2model/dataset_generator/__init__.py:4]: The dataset generator could probably also use an open-source model not through an online API, so maybe PromptBasedDatasetGenerator is OK?

File: prompt2model/dataset_generator/prompt_based.py (hunk 4 above)
Comment (neubig) [reply prompt2model/dataset_generator/prompt_based.py:83]: Seems so.

File: prompt2model/dataset_generator/prompt_based.py (hunk 5 above)
Comment (neubig) [reply prompt2model/dataset_generator/prompt_based.py:405]: Seems so: https://litellm.readthedocs.io/en/latest/input/

File: prompt2model/dataset_generator/readme.md (hunk 6 above)
Comment (neubig) [reply prompt2model/dataset_generator/readme.md:25]: It should be if you're using OpenAI, so I modified the doc a little.

File: prompt2model/utils/api_tools.py (hunk 7 above)
Comment (neubig) [reply prompt2model/utils/api_tools.py:18]: There's a mapping layer that maps all errors to openai exception types: https://github.com/BerriAI/litellm/commit/ef431415549dff307d1f312c32bdcbf43be3f3b4

**Comment by neubig** [comment]
Re: the main question, yes, lightllm automatically loads the API key from environment variables.

File: prompt2model/utils/api_tools.py (hunk 7 above)
Comment (zhaochenyang20) [reply prompt2model/utils/api_tools.py:18]: Sure! Should we add a comment here to explain this?

**Review by zhaochenyang20 — APPROVED** [review]
One last comment.

**Pull request merged**